### PR TITLE
new: basic CLI commands generation

### PR DESCRIPTION
**Pull request by t00f**
Autogenerate CLI commands based on an elemental.ModelManager.

### Usage
```
// Set up the cobra flags
ApoctlCmd.PersistentFlags().AddFlagSet(manipcli.ManipulatorFlagSet())
// Generate command
ApoctlCmd.AddCommand(manipcli.GenerateCobraCommand(gaia.Manager(), manipcli.ManipulatorMakerFromFlags()))
```

### Remaining tasks
- [x] Add missing commands (count, delete-many etc.)
- [x] Support parent relationship
- [x] Support templates
- [x] Unit tests

### CODE REVIEW

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "a3s",
    "component": "backend",
    "pr-id": "1448",
    "commit-sha": "21ac8c2d37cdc13ca138351a884100fccd8aecec",
    "pipeline": "master"
  },
  {
    "project": "a3s",
    "component": "elemental",
    "pr-id": "114",
    "commit-sha": "e264f153acc24c5d6a5c94cdec5433341787284c",
    "pipeline": "master"
  },
  {
    "project": "a3s",
    "component": "manipulate",
    "pr-id": "161",
    "commit-sha": "1cd089f256896b775154b3b80b88ad2c0bfd689b",
    "pipeline": "master"
  }
]
```

File: go.mod [hunk 1]
```diff
@@ -8,17 +8,34 @@ require (
...
 	github.com/smartystreets/goconvey v1.6.4
+	github.com/spf13/cobra v1.1.1
+	github.com/spf13/pflag v1.0.5
+	github.com/spf13/viper v1.7.1
 	go.uber.org/zap v1.19.0
+	golang.org/x/lint v0.0.0-20200130185559-910be7a94367 // indirect
 	golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
 	golang.org/x/sys v0.0.0-20210510120138-977fb7262007
+	k8s.io/helm v2.17.0+incompatible
+	sigs.k8s.io/yaml v1.3.0
```
Comment (primalmotion) [inline go.mod:40]: I don't think we want that or is it a dep from helm?

File: go.mod [hunk 2]
```diff
@@ -8,17 +8,34 @@ require (
...
 require (
+	github.com/Masterminds/goutils v1.1.0 // indirect
+	github.com/Masterminds/semver v1.5.0 // indirect
+	github.com/Masterminds/sprig v2.22.0+incompatible
+	github.com/araddon/dateparse v0.0.0-20200409225146-d820a6159ab1
+	github.com/ghodss/yaml v1.0.0
 	github.com/globalsign/mgo v0.0.0-20181015135952-eeefdecb41b8
 	github.com/gofrs/uuid v3.3.0+incompatible
 	github.com/golang/mock v1.6.0
+	github.com/google/uuid v1.1.1 // indirect
```
Comment (primalmotion) [inline go.mod:19]: check f this is necessary

File: go.mod [hunk 3]
```diff
@@ -8,17 +8,34 @@ require (
 )
 
 require (
+	github.com/Masterminds/goutils v1.1.0 // indirect
+	github.com/Masterminds/semver v1.5.0 // indirect
+	github.com/Masterminds/sprig v2.22.0+incompatible
+	github.com/araddon/dateparse v0.0.0-20200409225146-d820a6159ab1
+	github.com/ghodss/yaml v1.0.0
```
Comment (primalmotion) [inline go.mod:15]: double cvheck we don't already have a yaml package

File: manipcli/update.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"go.aporeto.io/elemental"
+	"go.aporeto.io/manipulate"
+	"sigs.k8s.io/yaml"
```
Comment (primalmotion) [inline manipcli/update.go:13]: use correct yaml package

File: manipcli/utils.go [hunk 5]
```diff
@@ -0,0 +1,707 @@
...
+
+	"github.com/Masterminds/sprig"
+	"github.com/araddon/dateparse"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"go.aporeto.io/elemental"
+	"go.aporeto.io/manipulate"
+	"go.uber.org/zap"
+	"k8s.io/helm/pkg/strvals"
+	"sigs.k8s.io/yaml"
```
Comment (primalmotion) [inline manipcli/utils.go:28]: use correct yaml package

File: Makefile [hunk 6]
```diff
@@ -13,7 +13,6 @@ lint:
 		--enable=errcheck \
 		--enable=goimports \
 		--enable=ineffassign \
-		--enable=revive \
```
Comment (primalmotion) [inline Makefile:16]: we want revive

**Review by primalmotion — CHANGES_REQUESTED** [review]
double check go mod. other than

File: go.mod (hunk 2 above)
Comment (t00f) [reply go.mod:19]: Looks like it is. It's a dependency from mastermind.

File: go.mod (hunk 3 above)
Comment (t00f) [reply go.mod:15]: This one seems correct. There is another yaml lib at the bottom which should be removed.

File: go.mod (hunk 1 above)
Comment (t00f) [reply go.mod:40]: Good catch, this is removed now

File: manipcli/update.go (hunk 4 above)
Comment (t00f) [reply manipcli/update.go:13]: Done

File: Makefile (hunk 6 above)
Comment (t00f) [reply Makefile]: Added

File: manipcli/utils.go (hunk 5 above)
Comment (t00f) [reply manipcli/utils.go]: Done

**Review by primalmotion — APPROVED** [review]
thx!

**Pull request merged**